tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
@odersky I wonder if we could somehow make something like https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/transform/Splitter.scala be a part of Typer to avoid having to select on a union type at all